presidio-analyzer/presidio_analyzer/analyzer_engine.py

```diff
@@ -55,30 +60,69 @@ def __init__(
             app_tracer = AppTracer()
         self.app_tracer = app_tracer
 
-        self.default_language = default_language
+        self.supported_languages = supported_languages
 
         self.nlp_engine = nlp_engine
         self.registry = registry
 
         # load all recognizers
         if not registry.recognizers:
-            registry.load_predefined_recognizers()
+            registry.load_predefined_recognizers(
+                nlp_engine=self.nlp_engine, languages=self.supported_languages
+            )
 
         self.enable_trace_pii = enable_trace_pii
         self.default_score_threshold = default_score_threshold
 
+    @staticmethod
+    def _create_nlp_engine(nlp_conf_file_path: Path) -> NlpEngine:
+        """
+        Create a specific NlpEngine class based on default configuration.
+
+        :param nlp_conf_file_path: Path to conf file
+        """
+        # Create specific NlpEngine given configuration file:
+        if not nlp_conf_file_path.exists():
+            nlp_configuration = {
+                "nlp_engine_name": "spacy",
+                "models": [{"lang_code": "en", "model_name": "en_core_web_lg"}],
+            }
+            logger.warning(
+                f"configuration file {NlpEngine.default_conf_file} not found.  "
+                f"Using default config: {nlp_configuration}."
+            )
+
+        else:
+            nlp_configuration = yaml.safe_load(open(nlp_conf_file_path))
+
+        nlp_engine = create_nlp_engine(nlp_configuration)
+        return nlp_engine
+
+    @staticmethod
+    def _get_full_conf_path() -> Path:
+        """Return a Path to the default conf file."""
+        return Path(Path(__file__).parent.parent, NlpEngine.default_conf_file)
+
     def get_recognizers(self, language: Optional[str] = None) -> List[EntityRecognizer]:
         """
         Return a list of PII recognizers currently loaded.
 
         :param language: Return the recognizers supporting a given language.
         :return: List of [Recognizer] as a RecognizersAllResponse
         """
-        if not language:
-            language = self.default_language
-        logger.info(f"Fetching all recognizers for language {language}")
-        recognizers = self.registry.get_recognizers(language=language, all_fields=True)
-        return recognizers
+        languages = self.supported_languages
+
+        if language:
```

Comment:
are you clearing self.supported_languages on purpose here? shouldn't this add to the list?

Reply:
No, good catch.

Reply:
not resolved yet?

Reply:
Fixed